### PR TITLE
don't show chat attachment dropdown if only one kind of file input is allowed

### PR DESCRIPTION
**Pull request by snopoke**
## Description
Resolves https://github.com/dimagi/open-chat-studio/issues/903

#### Tech note:
This adds [django-template-partials](https://github.com/carltongibson/django-template-partials?tab=readme-ov-file#basic-usage) library which can be used instead of 'include' in certain places.

## User Impact
Users don't have to click twice to upload files when only one kind of file is allowed.

### Demo
<div>
    <a href="https://www.loom.com/share/9cc7157966834047a252aafa4946efc7">
      <p>Dimagi Chatbots | Chat attachment dropdown changes - 22 November 2024 - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/9cc7157966834047a252aafa4946efc7">
      
    </a>
  </div>

### CODE REVIEW

File: templates/experiments/chat/input_bar.html [hunk 1]
```diff
@@ -20,55 +20,59 @@
...
+                class="file-input file-input-bordered file-input-sm w-full max-w-xs"
+              />
+            {% endpartialdef %}
+          </div>
+        {% elif assistant.supports_code_interpreter %}
+          <div role="button" class="btn m-1" x-bind:disabled="disableCodeInterpreterUpload"
+               @click="$refs.codeInterpreter.click()">
+            <i class="fa-solid fa-paperclip"></i>
+          </div>
+          {% partial code-interpreter-input %}
```
Comment (snopoke) [inline templates/experiments/chat/input_bar.html:70]: This tag renders the template partial defined above

File: templates/experiments/chat/input_bar.html [hunk 2]
```diff
@@ -20,55 +20,59 @@
...
+                <li>
+                  <a x-bind:disabled="disableCodeInterpreterUpload" @click="$refs.codeInterpreter.click()">Code
+                    Interpreter</a>
+                </li>
+                <li>
+                  <a @click="$refs.fileSearch.click()">File Search</a>
+                </li>
+              </ul>
+            </div>
+            {% partialdef code-interpreter-input inline=True %}
```
Comment (snopoke) [inline templates/experiments/chat/input_bar.html:38]: I'm using 'inline' partials so that they render here and can also be referenced elsewhere

Comment (stephherbers) [reply templates/experiments/chat/input_bar.html:38]: Nice so if without that flag the definition doesn't render it until called later on? Neat

**Review by stephherbers — APPROVED** [review]
I'm excited to use template-partials in other places 🌟

**Pull request merged**